service/history/handler.go

```diff
@@ -1295,6 +1295,44 @@ func (h *Handler) ReplicateRawEvents(
 	return nil
 }
 
+// ReplicateEventsV2 is called by processor to replicate history events for passive domains
+func (h *Handler) ReplicateEventsV2(
+	ctx context.Context,
+	replicateRequest *hist.ReplicateEventsV2Request,
+) (retError error) {
+
+	defer log.CapturePanic(h.GetLogger(), &retError)
+	h.startWG.Wait()
```

Comment:
What is this wait group for?

Reply:
https://github.com/uber/cadence/blob/master/service/history/handler.go#L195